File: javascript/workflow/src/verbs/impute.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */

import type { ImputeArgs } from '@datashaper/schema'

import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const imputeStep: ColumnTableStep<ImputeArgs> = (
	input,
	{ value, column },
) => {
	const dArgs = {
		[column]: (_d: any, $: any) => $.value,
	}

	return input.params({ value }).impute(dArgs)
}

export const impute = stepVerbFactory(imputeStep)

File: python/datashaper/datashaper/engine/verbs/impute.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Impute verb implementation."""

from typing import Any, cast

import pandas as pd

from datashaper.engine.verbs.verb_input import VerbInput
from datashaper.engine.verbs.verbs_mapping import verb
from datashaper.table_store.types import VerbResult, create_verb_result
from datashaper.verbs import impute


@verb(name="impute")
def impute_verb(input: VerbInput, **kwargs: Any) -> VerbResult:
    """Impute verb implementation."""
    result = impute(cast(pd.DataFrame, input.get_input()), **kwargs)
    return create_verb_result(result)
